include/aws/common/linked_list.h

```diff
@@ -210,10 +216,16 @@ AWS_STATIC_IMPL void aws_linked_list_insert_after(
 AWS_STATIC_IMPL void aws_linked_list_insert_before(
     struct aws_linked_list_node *before,
     struct aws_linked_list_node *to_add) {
+    AWS_PRECONDITION(aws_linked_list_node_prev_is_valid(before));
+    AWS_PRECONDITION(to_add != NULL);
```

Comment:
`to_add` doesn't need to be valid?

Reply:
`node_prev/next_is_valid` both check that the node's links are valid, but in this case, `to_add` was not part of the linked list, so its `next` and `prev` could point anywhere.